src/cljs/main/broadfcui/page/workspace/notebooks/notebooks_table.cljs

```diff
@@ -333,24 +333,24 @@
-                                      (if success?
-                                        (let [updated-notebooks (map (fn [n]
-                                                                       (if (= (:cluster-name n) (:clusterName cluster))
-                                                                         (merge n {:localized? true})
-                                                                         n))
-                                                                     notebooks)]
-                                          ; flip :localized? to true so we don't re-localize notebooks to this cluster
-                                          (swap! state assoc :server-response
-                                                 {:notebooks (map (fn [n]
-                                                                    (if (= (:cluster-name n) (:clusterName cluster))
-                                                                      (merge n {:localized? true})
-                                                                      n))
-                                                                  notebooks)})
-                                          (this :-persist-notebook-cluster-association updated-notebooks))
-                                        (js/setTimeout #(this :-localize-notebooks cluster) 10000))))))
+       (when (not-empty notebooks-to-localize)
+         (endpoints/localize-notebook (:googleProject cluster) (:clusterName cluster)
+                                      (reduce merge (conj (map (partial this :-localize-entry) notebooks-to-localize) (this :-delocalize-json)))
+                                      (fn [{:keys [success? get-parsed-response]}]
+                                        (when success?
+                                          (let [updated-notebooks (map (fn [n]
+                                                                         (if (= (:cluster-name n) (:clusterName cluster))
+                                                                           (merge n {:localized? true})
+                                                                           n))
+                                                                       notebooks)]
+                                            ; flip :localized? to true so we don't re-localize notebooks to this cluster
+                                            (swap! state assoc :server-response
+                                                   {:notebooks (map (fn [n]
+                                                                      (if (= (:cluster-name n) (:clusterName cluster))
+                                                                        (merge n {:localized? true})
+                                                                        n))
+                                                                    notebooks)})
```

Comment:
I see that this is not code you wrote, but isn't this a bunch of duplicated computation?

Reply:
Also `(assoc n :localized? true)` is preferred to the merge.

Reply:
hm, so it is. Will fix this